go/arrow/array/list_test.go

```diff
@@ -211,3 +211,190 @@ func TestListArraySlice(t *testing.T) {
 		t.Fatalf("got=%q, want=%q", got, want)
 	}
 }
+
+func TestLargeListArray(t *testing.T) {
+	pool := memory.NewCheckedAllocator(memory.NewGoAllocator())
+	defer pool.AssertSize(t, 0)
+
+	var (
+		vs      = []int32{0, 1, 2, 3, 4, 5, 6}
+		lengths = []int{3, 0, 4}
+		isValid = []bool{true, false, true}
+		offsets = []int64{0, 3, 3, 7}
```

Comment:
For the record, perhaps it would be nice to also test a zero-length non-null list element as well? e.g.:
```suggestion
		lengths = []int{3, 0, 0, 4}
		isValid = []bool{true, false, true, true}
		offsets = []int64{0, 3, 3, 3, 7}
```